test/AutomatedCarTests/SystemComponents/InputProcessTests.cs

```diff
@@ -0,0 +1,439 @@
+using System.Collections.Generic;
+using System.Text;
+using Xunit;
+
+namespace Tests.SystemComponents
+{
+    public class InputProcessTests : IDisposable
+    {
+        static HumanMachineInterface hmi;
+
+        public InputProcessTests()
+        {
+            hmi = new HumanMachineInterface(new VirtualFunctionBus());
+        }
+        
+        #region GearShiftTests
+        [Theory]
+        [InlineData(0, Gears.P)]
+        [InlineData(1, Gears.R)]
+        [InlineData(2, Gears.N)]
+        [InlineData(3, Gears.D)]
+        [InlineData(4, Gears.D)]
+        [InlineData(500, Gears.D)]
+        public void WithNumerousGearshiftsUpwardGearIncreasesTillDrivemode(int gearshiftsUp, Gears gear)
+        {
+            for (int i = 0; i < gearshiftsUp; i++)
+            {
+                hmi.GearCalculate(true, false);
+            }
+            Assert.Equal(gear, hmi.HmiPacket.Gear);
+        }
+
+        [Theory]
+        [InlineData(3, Gears.P)]
+        [InlineData(2, Gears.R)]
+        [InlineData(1, Gears.N)]
+        [InlineData(0, Gears.D)]
+        [InlineData(4, Gears.P)]
+        [InlineData(500, Gears.P)]
+        public void WithNumerousGearshiftsDownwardGearDecreasesTillParkingmode(int gearshiftsDown, Gears gear)
+        {
+            ((HMIPacket)hmi.HmiPacket).Gear = Gears.D;
+            for (int i = 0; i < gearshiftsDown; i++)
+            {
+                hmi.GearCalculate(false, true);
+            }
+            Assert.Equal(gear, ((HMIPacket)hmi.HmiPacket).Gear);
+        }
+
+        [Theory]
+        [InlineData(10, Gears.P)]
+        [InlineData(10, Gears.R)]
+        [InlineData(10, Gears.N)]
+        [InlineData(10, Gears.D)]
+        public void WithBothGearUpAndGearDownBeingTrueGearDoesentChangeItsValue(int indx, Gears gear)
+        {
+            ((HMIPacket)hmi.HmiPacket).Gear = gear;
+            for (int i = 0; i < indx; i++)
+            {
+                hmi.GearCalculate(true, true);
+                Assert.Equal(gear, ((HMIPacket)hmi.HmiPacket).Gear);
+            }
+        }
+
+        [Theory]
+        [InlineData(10, Gears.P)]
+        [InlineData(10, Gears.R)]
+        [InlineData(10, Gears.N)]
+        [InlineData(10, Gears.D)]
+        public void WithBothGearUpAndGearDownBeingFalseGearDoesentChangeItsValue(int indx, Gears gear)
+        {
+            ((HMIPacket)hmi.HmiPacket).Gear = gear;
+            for (int i = 0; i < indx; i++)
+            {
+                hmi.GearCalculate(false, false);
+                Assert.Equal(gear, ((HMIPacket)hmi.HmiPacket).Gear);
+            }
+        }
+
+        #endregion
+
+        #region AccTests
+
+        [Fact]
+        public void WithNoInputAccSetDoesentChangeTheStateOfAcc()
+        {
+            var acc = ((HMIPacket)hmi.HmiPacket).Acc;
+            hmi.AccSet(false);
+
+            Assert.Equal(acc, ((HMIPacket)hmi.HmiPacket).Acc);
+        }
+
+        [Fact]
+        public void WithNoInputAccSpeedSetDoesentChangeTheDesiredSpeed()
```

Comment:
Ezt is írtam már egy másiknál, hogy Does**e**nt nincs. DoesNot vagy Doesnt...

Reply:
igen csak az egy másik branch-en lett kijavítva ami nem lett még be mergelve :/

Reply:
Igen, elrontottam a merge sorrendet, ami egyéb gondot is okoz a másiknál.